AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -160,9 +152,32 @@ - (void)_batchLayoutNodesFromContexts:(NSArray<ASIndexedNodeContext *> *)context
   }];
 }
 
+/**
+ * Perform measurement and layout of loaded or unloaded nodes based if they will be layed out on main thread or not
+ */
+- (void)_layoutNodes:(NSArray<ASCellNode *> *)nodes onMainThread:(BOOL)onMainThread fromContexts:(NSArray<ASIndexedNodeContext *> *)contexts ofKind:(NSString *)kind
```

Comment:
I'm wondering if we can remove `onMainThread` parameter? Instead, we can check `node.isNodeLoaded == NSThread.isMainThread` and only layout a particular node if it passes this test. With that, this method measures loaded nodes if it is called on main thread and unloaded ones if it is called on a background thread. This is exactly the way we are using it.